sdk-wallet/user_wallet_provider.md

```diff
@@ -5,6 +5,14 @@ class UserWalletProvider:
     // Should not throw.
     generate_keypair(): (ISecretKey, IPublicKey)
 
+    // Should not throw (generally speaking).
+    //
+    // Generates a keypair that satisfies the given constraints.
+    // 
+    // Generally speaking, the input paramters would be a predicate (function that returns a boolean) 
+    // or something similar (depending on the programming language).
+    generate_keypair_with_constraint(isConstraintSatisfied: Predicate[input = (ISecretKey, IPublicKey), output = boolean]): (ISecretKey, IPublicKey)
```

Comment:
Or perhaps this should be optional, not part of specs?

Reply:
Yes, I think so, I would keep the specs simple and to the point

Reply:
All right, removed `generate_keypair_with_constraint` from specs.